src/components/NewMovie/NewMovie.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { Component } from 'react';
+import propTypes from 'prop-types';
+import Grid from '@material-ui/core/Grid';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+
+// eslint-disable-next-line
+const regex = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/
+
+export class NewMovie extends Component {
+  state = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+    imgValid: true,
+    imdbValid: true,
+  };
+
+  validateImgUrl = () => (
+    this.setState(prevState => ({
+      imgValid: regex.test(prevState.imgUrl),
+    }))
+  )
+
+  validateImdbUrl = () => (
+    this.setState(prevState => ({
+      imdbValid: regex.test(prevState.imdbUrl),
+    }))
+  )
+
+  handleChange = (e) => {
+    const { name, value } = e.target;
+
+    this.setState({
+      [name]: value,
+    });
+
+    if (name === 'imgUrl') {
+      this.validateImgUrl();
+    }
+
+    if (name === 'imdbUrl') {
+      this.validateImdbUrl();
+    }
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { onAdd } = this.props;
+    const {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    } = this.state;
+
+    const newMovie = {
+      title,
+      description,
+      imdbId,
+      imdbUrl,
+      imgUrl,
+    };
+
+    onAdd(newMovie);
```

Comment:
better name for the function
```suggestion
    addMovie(newMovie);
```